lms/services/canvas_api/__init__.py

```diff
@@ -0,0 +1,36 @@
+from urllib.parse import urlparse
+
+from lms.services.canvas_api.authenticated import AuthenticatedClient
+from lms.services.canvas_api.basic import BasicClient
+from lms.services.canvas_api.public import CanvasAPIClient
+from lms.services.canvas_api.token_store import TokenStore
+
+
+def canvas_api_client_factory(_context, request):
+    """
+    Get a CanvasAPIClient from a pyramid request.
+
+    :param _context: Pyramid context object
+    :param request: Pyramid request object
+    :return: An instance of CanvasAPIClient
+    """
```

Comment:
Not sure how I feel about this completely. It 100% decouples Pyramid from the classes underneath, which is good. But it's super ugly stuff.

I think it's probably better to keep the ugly in one place on balance. It's also good documentation for how the objects all fit together.

Reply:
Yeah I think it's probably good to do it this way